### DISCUSSION TITLE
Conversion to grayscale - but for (scanned) documnets

### DISCUSSION THREAD

**Discussion by halloleo** [ImageMagick/ImageMagick#6282] [Help]
### ImageMagick version

7.1.0-51 

### Operating system, version and so on

macOS 10.15

### Description

I have lots of scannes which i tool as color mode scanns (becasue in color mode does teh OCR work best).

Here an example:

![Warranty card_Color](https://user-images.githubusercontent.com/1521852/234286302-0a4b65d8-4746-40f7-b4a2-a666764ff0be.jpg)

For archiving  I want to convert the images to grayscale, but so that the white of the page background  is really white and the black of the text is quite dark but the text outlines are still smooth.

I tried 

* IM's `-monochrome`
* IM's `-threshold` with various threshold
* `pamthreshold`
But the result looks always something like this:

![Warranty card_bw-pam](https://user-images.githubusercontent.com/1521852/234288500-92f68d47-247b-49cf-9ca0-4d4143fe8868.jpg)

This is not at all like the physical document I have in my hand! The document is black on white, but with smooth font strokes -- how can I do a better conversion?

Thanks for any pointers!

**Comment by GeeMack** [comment]
Maybe try using "-level" to adjust the black and white range. Here's a test command...

    magick input.png -level 40,90% result.png

... and the result...

![result](https://user-images.githubusercontent.com/65475279/234305408-5721c818-f28e-4884-a578-436d4b350f55.png)

**Reply by halloleo** [reply]
Great idea!

Also thanks for providing an result image example!

**Comment by snibgo** [comment]
> ... so that the white of the page background is really white and the black of the text is quite dark but the text outlines are still smooth.

An alternative is to stretch the tones so about 10% become black and 70% become white, with grays between them, for example:
```
magick warranty.jpg -contrast-stretch 10x70% x.png
```

**Reply by halloleo** [reply]
Looks promising. Thanks. Will try it out.

**Reply by halloleo** [reply]
Ok, I have tried this - and it works. However to get reasonably sharp results with PDFs I need to bump up the density to 600dpi! So I have 

    magick -density 600dpi 'Warranty.pdf' -contrast-stretch 10x70% 'Warranty_bw.pdf'

Then Warranty_bw.pdf looks pretty good (i.e. similarly sharp as the original) but the 200KB original gets blown up to 43MB!!! Any idea how to keep it sharp but with a more reasonable file size?

**Comment by fmw42** [comment]
Use -density to get high resolution and scale up, then resize by the inverse amount (or to whatever size you want). Default density is 72.  72*8=576. So resize by 1/8=12.5%

`magick -density 576 -units pixelsperinch 'Warranty.pdf' -resize 12.5% -contrast-stretch 10x70% 'Warranty_bw.pdf'`
